src/core/server/saved_objects/migrationsv2/integration_tests/migration_from_older_v1.test.ts

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import Path from 'path';
+import Fs from 'fs';
+import Util from 'util';
+import Semver from 'semver';
+import { REPO_ROOT } from '@kbn/dev-utils';
+import { Env } from '@kbn/config';
+import { getEnvOptions } from '../../../config/mocks';
+import * as kbnTestServer from '../../../../test_helpers/kbn_server';
+import { ElasticsearchClient } from '../../../elasticsearch';
+import { SavedObjectsRawDoc } from '../../serialization';
+import { InternalCoreStart } from '../../../internal_types';
+import { Root } from '../../../root';
+
+const kibanaVersion = Env.createDefault(REPO_ROOT, getEnvOptions()).packageInfo.version;
+
+const logFilePath = Path.join(__dirname, 'migration_from_older_v1.log');
+
+const asyncUnlink = Util.promisify(Fs.unlink);
+async function removeLogFile() {
+  // ignore errors if it doesn't exist
+  await asyncUnlink(logFilePath).catch(() => void 0);
+}
+const assertMigratedDocuments = (arr: any[], target: any[]) => target.every((v) => arr.includes(v));
+
+function sortByTypeAndId(a: { type: string; id: string }, b: { type: string; id: string }) {
+  return a.type.localeCompare(b.type) || a.id.localeCompare(b.id);
+}
+
+async function fetchDocuments(esClient: ElasticsearchClient, index: string) {
+  const { body } = await esClient.search<any>({
+    index,
+    body: {
+      query: {
+        match_all: {},
+      },
+      _source: ['type', 'id'],
+    },
+  });
+
+  return body.hits.hits
+    .map((h) => ({
+      ...h._source,
+      id: h._id,
+    }))
+    .sort(sortByTypeAndId);
+}
+
+describe('migration v2', () => {
```

Comment:
maybe we should remove the skipped test? https://github.com/elastic/kibana/pull/112260/files#diff-3481989185237dc1618f58d58df3393585820b958c5682512d6bb1f857710a5eR163